### PR TITLE
pessimistictest: dedicated package for pessimistic txn tests

### LINKED ISSUE #28325: migrate test-infra to testify for `session/pessimistic_test.go`

### PR DESCRIPTION
**Pull request by tisonkun**
Issue Number: close #28325 

```release-note
None
```

### CODE REVIEW

File: .github/workflows/pessimistic_test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Setup TiKV cluster
+        run: |
+          # Disable pipelined pessimistic lock temporarily until tikv#11649 is resolved
+          echo -e "[pessimistic-txn]\npipelined = false" > tikv.toml
+          curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | cat
```
Comment (tisonkun) [inline .github/workflows/pessimistic_test.yml:21]:
```suggestion
          curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
```

File: .github/workflows/pessimistic_test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Setup TiKV cluster
+        run: |
+          # Disable pipelined pessimistic lock temporarily until tikv#11649 is resolved
+          echo -e "[pessimistic-txn]\npipelined = false" > tikv.toml
+          curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
+          tiup playground nightly --mode tikv-slim --kv 3 --pd 1 --without-monitor --kv.config tikv.toml
```
Comment (tisonkun) [inline .github/workflows/pessimistic_test.yml:22]:
```suggestion
          $HOME/.tiup/bin/tiup playground nightly --mode tikv-slim --kv 3 --pd 1 --without-monitor --kv.config tikv.toml
```

File: pessimistictest/pessimistic_test.go [hunk 3]
```diff
@@ -12,51 +12,167 @@
...
+	tk := testkit.NewTestKit(t, store)
+	tk.Session().GetSessionVars().EnableAsyncCommit = true
+	tk.MustExec("use test")
 	return tk
 }
 
-func (s *testPessimisticSuite) new1PCTestKitWithInit(c *C) *testkit.TestKit {
-	tk := testkit.NewTestKitWithInit(c, s.store)
-	tk.Se.GetSessionVars().Enable1PC = true
+func create1PCTestKit(t *testing.T, store kv.Storage) *testkit.TestKit {
```
Comment (tisonkun) [inline pessimistictest/pessimistic_test.go:172]:
```suggestion
// TODO: figure out a stable way to run Test1PCWithSchemaChange
//nolint:unused
func create1PCTestKit(t *testing.T, store kv.Storage) *testkit.TestKit {
```

**Review by hawkingrei — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by xhebox** [comment]
LGTM, but I am not sure whether this should be reviewed by other reviewers.

**Comment by tisonkun** [comment]
@xhebox @hawkingrei interesting. There is no `status/LGT2` label.

**Review by hawkingrei — APPROVED** [review]

File: pessimistictest/pessimistic_test.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 PingCAP, Inc.
```
Comment (hawkingrei) [inline pessimistictest/pessimistic_test.go:1]: Can this test be moved to ```tests `` folders?

**Review by hawkingrei — APPROVED** [review]

File: pessimistictest/pessimistic_test.go (hunk 4 above)
Comment (bb7133) [reply pessimistictest/pessimistic_test.go]: Is there any specific reason we would like to do the move? The current tests in `tests` folds are those of 'integration tests'.

File: tests/pessimistictest/pessimistic_test.go [hunk 5]
```diff
@@ -12,51 +12,169 @@
...
 
-func (s *testPessimisticSuite) newAsyncCommitTestKitWithInit(c *C) *testkit.TestKit {
-	tk := testkit.NewTestKitWithInit(c, s.store)
-	tk.Se.GetSessionVars().EnableAsyncCommit = true
+func TestMain(m *testing.M) {
+	testbridge.SetupForCommonTest()
+	flag.Parse()
+	session.SetSchemaLease(20 * time.Millisecond)
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.TiKVClient.AsyncCommit.SafeWindow = 0
```
Comment (bb7133) [inline tests/pessimistictest/pessimistic_test.go:68]: Could we confirm that making `SafeWindow=0` is a fine default value for all the tests?

Comment (tisonkun) [reply tests/pessimistictest/pessimistic_test.go:68]: This is the setting from `seesion/main_test.go`. I tried to avoid unnecessary changes.

**Comment by tisonkun** [comment]
@hawkingrei it seems @ti-chi-bot still not work so even we pass unit-test we cannot merge this pr :)

**Comment by hawkingrei** [comment]
> @hawkingrei it seems @ti-chi-bot still not work so even we pass unit-test we cannot merge this pr :)

I have applied for repairs, waiting for EE feedback. : (

**Review by hawkingrei — APPROVED** [review]

**Comment by tisonkun** [comment]
@xhebox @bb7133 please git this PR another review >_<

**Comment by tisonkun** [comment]
I think @ti-chi-bot has just revived.

**Review by xhebox — APPROVED** [review]

**Pull request merged**